### ISSUE TITLE
Rewrite UpdateRowTrafo and similar functions

### ISSUE THREAD

**Issue by emmakatherina** [lbfm-rwth/GaussPar#52]
Functions like UpdateRowTrafo should:
- get an option whether to run in parallel or sequential mode
- take a ¨result¨ variable into which it writes

If working in sequential mode it directly writes results into the variable. If working in parallel mode it writes task objects.

**Comment by emmakatherina** [comment]
Also, it would be nice if UpdateRowTrafo could schedule a task that performs the computation and also schedules tasks that write the result into the correct list.
So that we can get rid of the following code in the main thread:
```
                     K[i][h] := TaskResult( TaskListUpdateM[i][j][h] ).K;
                     M[j][h] := TaskResult( TaskListUpdateM[i][j][h] ).M;
```

**Comment by jnkbrachter** [comment]
The option is no longer needed, since all these functions were replaced by destructive versions everywhere.
This also takes care of the second point.
